crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/AbstractLockCommand.java

```diff
@@ -98,13 +103,19 @@ public CallStatus run() {
             o.set("result", result);
             return new CallStatus(new SimpleJsonObject(o));
         } catch (Error e) {
+            metricCtx.markRequestException(e, e.getErrorCode());
             LOGGER.error("failure: {}", e);
             return new CallStatus(e);
         } catch (Exception e) {
+            metricCtx.markRequestException(e, e.getMessage());
             LOGGER.error("failure: {}", e);
             return new CallStatus(Error.get(RestCrudConstants.ERR_REST_ERROR, e.toString()));
+        } finally {
+            metricCtx.endRequestMonitoring();
         }
     }
 
     protected abstract JsonNode runLockCommand(Locking locking);
+    
+    public abstract String getLockCommandName();
```

Comment:
Would probably keep this protected if possible FWIW

Reply:
Changed visibility.